### PR TITLE
[FIX] l10n_nl: set 'Cost of Revenue' in allowed account types for Ven…

### PR DESCRIPTION
**Pull request by hubvd**
…dor Bills journal created for a l10n_nl chart of account

Allow Cost of Revenue accounts to match the default expense account

Ticket: 2458146

### CODE REVIEW

File: addons/l10n_nl/models/account_journal.py [hunk 1]
```diff
@@ -6,6 +6,19 @@
...
     _inherit = 'account.journal'
 
+    @api.model
+    def _fill_missing_values(self, vals):
+        super()._fill_missing_values(vals)
+
+        if vals.get('type') != 'purchase':
+            return
+
+        company = self.env['res.company'].browse(vals['company_id'])
```
Comment (william-andre) [inline addons/l10n_nl/models/account_journal.py:16]: You can't be sure `company_id` is the the values

File: addons/l10n_nl/models/account_journal.py [hunk 2]
```diff
@@ -6,6 +6,19 @@
...
+        if company.country_id.code == "NL":
+            cost_of_revenue_account_type = self.env["account.account.type"].search([("name", "=", "Cost of Revenue")])
```
Comment (william-andre) [inline addons/l10n_nl/models/account_journal.py:18]: Get it by xmlid instead `account.data_account_type_direct_costs`

File: addons/l10n_nl/models/account_journal.py [hunk 3]
```diff
@@ -6,6 +6,19 @@
...
+            if cost_of_revenue_account_type and len(self.type_control_ids) == 0:
```
Comment (william-andre) [inline addons/l10n_nl/models/account_journal.py:19]: You are in a hook called in `create` before the `create` is actually done: `self.field` will always be False/empty/0

**Review by william-andre — CHANGES_REQUESTED** [review]

File: addons/l10n_nl/models/account_journal.py (hunk 1 above)
Comment (hubvd) [reply addons/l10n_nl/models/account_journal.py:16]: If the company is not present in the values, can I use self.env.company instead ?

Comment (hubvd) [reply addons/l10n_nl/models/account_journal.py:16]: Nevermind, I saw how to do it in the parent model.

**Comment by hubvd** [comment]
@robodoo retry

**Comment by hubvd** [comment]
@william-andre Everything should be OK now.

File: addons/l10n_nl/models/account_journal.py [hunk 4]
```diff
@@ -6,6 +6,17 @@
...
+    @api.model
+    def _fill_missing_values(self, vals):
+        super()._fill_missing_values(vals)
+
+        if vals.get('type') != 'purchase':
+            return
+
+        company = self.env['res.company'].browse(vals['company_id']) if vals.get('company_id') else self.env.company
+        if company.country_id.code == "NL":
+            vals['type_control_ids'] = [self.env.ref('account.data_account_type_direct_costs').id]
```
Comment (william-andre) [inline addons/l10n_nl/models/account_journal.py:18]: If there are already values for this key, you should add them to the id you are adding

**Review by william-andre — CHANGES_REQUESTED** [review]
The `retry` command is for when the CI did not pass the test on mergebot; most likely because of non deterministic tests (:warning: that are not related to your changes) or because mergebot was down for some reason (tests in the CI, changes of server configs, hardware failure,...)

File: addons/l10n_nl/models/account_journal.py (hunk 4 above)
Comment (hubvd) [reply addons/l10n_nl/models/account_journal.py]: Done

**Comment by william-andre** [comment]
Except that it would be better to use the [ORM commands](https://www.odoo.com/documentation/14.0/developer/reference/addons/orm.html#odoo.models.Model.write) (only the command 6 should be fine)

**Review by william-andre — APPROVED** [review]
Seems like it is indeed the right account type, and changing it to have expenses was not an option

@robodoo  r+

**Pull request closed without merging**